# Executive Summary  
## Customer Acquisition & Growth Analytics — US Credit Card Business

---

## Business Context

A US-based credit card issuer seeks to improve acquisition efficiency and long-term growth by understanding how marketing investment, acquisition channels, and customer segments translate into activated customers and early revenue generation.

While acquisition volumes remain healthy, leadership lacks a clear, data-driven view of where customers drop off, which channels are cost-efficient, and which sources deliver high-quality customers.

---

## Objective

The objective of this analysis was to evaluate the **end-to-end customer acquisition funnel**, assess **marketing spend efficiency**, and measure **early customer value**, in order to support value-driven acquisition strategy decisions.

---

## Key Insights

- **Activation is the primary funnel bottleneck**  
  ~70% of applications are approved, but only ~53% of applicants activate, indicating post-approval onboarding friction as the largest opportunity for improvement.

- **Funnel performance is consistent, but cost efficiency is not**  
  Approval and activation rates are broadly similar across channels, yet Customer Acquisition Cost (CAC) varies significantly, driven by differences in marketing spend intensity.

- **Digital acquisition delivers the lowest CAC at scale**  
  Digital channels achieve the most favorable balance between scale and cost efficiency, making them the strongest driver of acquisition volume.

- **Low CAC does not always imply highest customer value**  
  Partner channels, despite higher CAC, generate higher average and median early-stage revenue, highlighting a trade-off between cost efficiency and customer quality.

- **Early customer value is unevenly distributed**  
  A subset of customers and segments contributes disproportionately to early revenue, reinforcing the importance of value-based acquisition optimization.

---

## Strategic Recommendations

1. **Prioritize activation optimization**  
   Focus on improving post-approval onboarding experiences, as even small gains in activation rate can materially increase activated customers without additional marketing spend.

2. **Adopt a value-adjusted acquisition strategy**  
   Balance CAC with early customer value by prioritizing channels and segments that maximize early LTV relative to acquisition cost.

3. **Reallocate spend toward efficient growth drivers**  
   Incrementally shift budget toward channels with strong activation performance and competitive CAC, while optimizing or reassessing high-cost channels.

4. **Target high-value customer segments**  
   Refine targeting strategies to attract segments with higher revenue potential, even if they contribute lower acquisition volume.

---

## Business Impact

By shifting focus from volume-driven acquisition to **value-driven optimization**, the organization can increase the number of revenue-generating customers, improve marketing ROI, and support more sustainable long-term growth — without increasing total acquisition spend.

---


# Customer Acquisition & Growth Analytics  
## US Credit Card Business

---

## 1. Business Background

A US-based credit card issuer acquires customers through multiple channels including digital marketing, referrals, and strategic partnerships. While acquisition volumes have increased over time, leadership lacks clear visibility into which channels and customer segments generate high-quality, profitable customers.

The analytics team supports growth strategy by evaluating acquisition performance, customer behavior, and revenue outcomes. This analysis is intended to enable data-driven decisions around marketing investment, customer targeting, and onboarding optimization.

---

## 2. Problem Statement

The objective of this analysis is to evaluate customer acquisition performance across channels and customer segments in order to identify opportunities to improve conversion efficiency, customer quality, and early-stage revenue outcomes.

---

## 3. Key Business Questions

1. Which acquisition channels deliver the highest application-to-approval and activation conversion rates?
2. How does customer quality differ across acquisition channels and customer segments?
3. What is the relationship between customer acquisition cost and early revenue generation?
4. Which customer cohorts demonstrate stronger early retention and usage behavior?
5. What data-driven actions can leadership take to optimize acquisition strategy and marketing investment?

---

## 4. Key Performance Indicators (KPIs)

| KPI | Definition | Business Importance |
|---|---|---|
| Application Conversion Rate | Approved customers / Total applications | Measures acquisition funnel efficiency |
| Activation Rate | Activated customers / Approved customers | Indicates onboarding and early engagement effectiveness |
| Customer Acquisition Cost (CAC) | Marketing spend / Activated customers | Evaluates cost efficiency of acquisition channels |
| Revenue per Customer | Total revenue / Active customers | Measures customer value and quality |
| 90-Day Retention Rate | Customers active after 90 days / Activated customers | Early indicator of long-term customer value |

---

## 5. Scope and Assumptions

- The analysis covers acquisition and customer behavior data for the most recent six-month period.
- Revenue metrics represent early-stage customer value based on transaction activity and fees.
- The dataset is assumed to be anonymized and representative of real-world credit card acquisition behavior.
- Results are intended to support strategic decision-making rather than exact financial forecasting.

---


# Data Design & Schema Definition

---

## 1. Data Overview

To evaluate customer acquisition performance, customer quality, and revenue outcomes, the analysis uses a structured relational dataset representing the end-to-end acquisition and early customer lifecycle.

The dataset is designed to mirror how acquisition, customer, and marketing data are typically stored in financial services organizations.

---

## 2. Data Tables Overview

| Table Name | Description |
|-----------|------------|
| customers | Customer demographic attributes and acquisition source |
| applications | Credit card application and approval funnel |
| activations | Card activation and onboarding events |
| transactions | Customer transaction behavior |
| revenue | Revenue generated from customer activity |
| marketing_spend | Channel-level marketing investment |

---

## 3. Table Schemas

### customers

| Column | Description |
|------|------------|
| customer_id | Unique customer identifier |
| acquisition_channel | Channel through which the customer was acquired |
| segment | Customer segment (e.g., Mass, Affluent, SME) |
| state | US state of residence |
| signup_date | Date customer entered the acquisition funnel |

---

### applications

| Column | Description |
|------|------------|
| application_id | Unique application identifier |
| customer_id | Customer identifier |
| application_date | Date of application |
| application_status | Approved / Rejected |
| approval_date | Date of approval (if approved) |

---

### activations

| Column | Description |
|------|------------|
| customer_id | Customer identifier |
| activation_date | Card activation date |
| activation_status | Activated / Not Activated |

---

### transactions

| Column | Description |
|------|------------|
| transaction_id | Unique transaction identifier |
| customer_id | Customer identifier |
| transaction_date | Date of transaction |
| transaction_amount | Transaction value |
| merchant_category | Merchant category code |

---

### revenue

| Column | Description |
|------|------------|
| customer_id | Customer identifier |
| revenue_date | Date revenue was generated |
| revenue_amount | Revenue value (fees, interest, interchange) |

---

### marketing_spend

| Column | Description |
|------|------------|
| channel | Acquisition channel |
| spend_date | Date of marketing spend |
| spend_amount | Marketing spend amount |

---

## 4. Entity Relationships

- One customer may submit one application
- Approved applications may result in card activation
- Activated customers generate transactions and revenue
- Marketing spend is associated with acquisition channels

This relational structure enables funnel analysis, customer-level performance tracking, and channel ROI evaluation.

---

## 5. Data Design Rationale

This schema enables:
- End-to-end acquisition funnel analysis
- Channel-level cost and ROI comparison
- Customer segmentation and cohort tracking
- Early-stage revenue and retention measurement

The design supports scalable analytics while remaining simple enough for exploratory analysis in a notebook-based environment.

---


# Synthetic Data Generation

---

## 1. Purpose of Synthetic Data

To simulate real-world credit card acquisition analytics, this project uses large-scale synthetic data designed to reflect realistic customer behavior, acquisition funnels, and revenue patterns.

The dataset intentionally includes noise, missing values, and inconsistencies to replicate the data quality challenges commonly encountered in production analytics environments.

---

## 2. Data Volume Overview

| Table | Approximate Rows |
|-----|------------------|
| customers | 150,000 |
| applications | 150,000 |
| activations | ~105,000 |
| transactions | ~600,000 |
| revenue | ~300,000 |
| marketing_spend | ~3,000 |

This scale enables performance analysis, aggregation challenges, and realistic KPI behavior.

---

## 3. Realism & Noise Design

The data intentionally includes:
- Rejected applications
- Non-activated approved customers
- Skewed transaction amounts
- Missing demographic values
- Delayed activation behavior
- Channel-level performance differences

These characteristics ensure the analysis reflects real-world ambiguity rather than idealized outcomes.

---


In [ ]:
# Import Libraries
import numpy as np
import pandas as pd

np.random.seed(42)


In [ ]:
# Generate Customers Table

n_customers = 150_000

customers = pd.DataFrame({
    "customer_id": np.arange(1, n_customers + 1),
    "acquisition_channel": np.random.choice(
        ["Digital", "Referral", "Partner", "Organic"],
        size=n_customers,
        p=[0.45, 0.25, 0.20, 0.10]
    ),
    "segment": np.random.choice(
        ["Mass", "Affluent", "SME"],
        size=n_customers,
        p=[0.65, 0.25, 0.10]
    ),
    "state": np.random.choice(
        ["CA", "NY", "TX", "FL", "IL", "PA", "OH", "GA", "NC", "MI"],
        size=n_customers
    ),
    "signup_date": pd.to_datetime("2024-01-01") +
        pd.to_timedelta(np.random.randint(0, 180, size=n_customers), unit="D")
})

# Introduce missing values (realistic noise)
mask = np.random.rand(n_customers) < 0.03
customers.loc[mask, "segment"] = None

customers.head()


,customer_id,acquisition_channel,segment,state,signup_date
0,1,Digital,Mass,TX,2024-06-06
1,2,Organic,Mass,FL,2024-04-09
2,3,Partner,Affluent,CA,2024-04-27
3,4,Referral,Affluent,TX,2024-02-01
4,5,Digital,Affluent,IL,2024-03-17


In [ ]:
# Generate Applications Table

applications = customers[["customer_id", "signup_date"]].copy()
applications["application_id"] = np.arange(1, n_customers + 1)

applications["application_status"] = np.random.choice(
    ["Approved", "Rejected"],
    size=n_customers,
    p=[0.70, 0.30]
)

applications["approval_date"] = pd.NaT

approved_mask = applications["application_status"] == "Approved"

applications.loc[approved_mask, "approval_date"] = (
    applications.loc[approved_mask, "signup_date"]
    + pd.to_timedelta(
        np.random.randint(1, 7, size=approved_mask.sum()),
        unit="D"
    )
)


applications.rename(columns={"signup_date": "application_date"}, inplace=True)
applications.head()


,customer_id,application_date,application_id,application_status,approval_date
0,1,2024-06-06,1,Approved,2024-06-11
1,2,2024-04-09,2,Approved,2024-04-12
2,3,2024-04-27,3,Approved,2024-05-02
3,4,2024-02-01,4,Approved,2024-02-07
4,5,2024-03-17,5,Approved,2024-03-22


In [ ]:
# ACTIVATIONS TABLE

approved = applications[applications["application_status"] == "Approved"].copy()
approved = approved.reset_index(drop=True)

activation_prob = 0.75
activation_flags = np.random.rand(len(approved)) < activation_prob

activations = pd.DataFrame({
    "customer_id": approved["customer_id"],
    "activation_status": np.where(activation_flags, "Activated", "Not Activated"),
    "activation_date": pd.NaT
})

# Assign activation dates ONLY for activated customers
activated_idx = activations.index[activations["activation_status"] == "Activated"]

activations.loc[activated_idx, "activation_date"] = (
    approved.loc[activated_idx, "approval_date"]
    + pd.to_timedelta(
        np.random.randint(1, 14, size=len(activated_idx)),
        unit="D"
    )
)

activations.head()


,customer_id,activation_status,activation_date
0,1,Activated,2024-06-13
1,2,Activated,2024-04-21
2,3,Activated,2024-05-08
3,4,Not Activated,NaT
4,5,Activated,2024-04-01


In [ ]:
# TRANSACTIONS TABLE

active_customers = activations.loc[
    activations["activation_status"] == "Activated", "customer_id"
]

n_transactions = 600_000

transactions = pd.DataFrame({
    "transaction_id": np.arange(1, n_transactions + 1),
    "customer_id": np.random.choice(active_customers, size=n_transactions),
    "transaction_date": pd.to_datetime("2024-01-01")
        + pd.to_timedelta(np.random.randint(0, 180, size=n_transactions), unit="D"),
    "transaction_amount": np.round(
        np.random.lognormal(mean=3.5, sigma=1.0, size=n_transactions), 2
    ),
    "merchant_category": np.random.choice(
        ["Retail", "Travel", "Dining", "Fuel", "E-commerce"],
        size=n_transactions,
        p=[0.30, 0.25, 0.20, 0.15, 0.10]
    )
})

# Inject realistic noise: zero & extreme transactions
noise_mask = np.random.rand(n_transactions) < 0.01
transactions.loc[noise_mask, "transaction_amount"] *= np.random.choice([0, 5], size=noise_mask.sum())

transactions.head()


,transaction_id,customer_id,transaction_date,transaction_amount,merchant_category
0,1,127166,2024-05-02,5.01,Travel
1,2,46562,2024-06-08,119.90,Retail
2,3,139626,2024-01-12,7.23,Dining
3,4,38348,2024-02-16,5.35,Dining
4,5,49476,2024-02-26,22.22,Fuel


In [ ]:
# Generate Revenue Table

# REVENUE TABLE (derived from transactions)

revenue = transactions.sample(frac=0.5, random_state=42).copy()

revenue["revenue_date"] = revenue["transaction_date"]

# Revenue as a % of transaction value (interchange + fees simulation)
revenue["revenue_amount"] = np.round(
    revenue["transaction_amount"] * np.random.uniform(0.01, 0.03, size=len(revenue)),
    2
)

# Introduce missing revenue values (real-world noise)
missing_rev_mask = np.random.rand(len(revenue)) < 0.02
revenue.loc[missing_rev_mask, "revenue_amount"] = np.nan

revenue = revenue[["customer_id", "revenue_date", "revenue_amount"]]

revenue.head()


,customer_id,revenue_date,revenue_amount
4242,85572,2024-01-22,0.37
60608,68745,2024-01-29,1.58
392832,90603,2024-05-29,0.15
41643,108066,2024-05-25,0.12
464234,110588,2024-01-11,0.79


In [ ]:
# MARKETING SPEND TABLE

dates = pd.date_range(start="2024-01-01", periods=180, freq="D")
channels = ["Digital", "Referral", "Partner", "Organic"]

marketing_spend = pd.DataFrame({
    "channel": np.repeat(channels, len(dates)),
    "spend_date": np.tile(dates, len(channels)),
    "spend_amount": np.round(
        np.random.uniform(5_000, 50_000, size=len(dates) * len(channels)),
        2
    )
})

# Channel-specific spend bias (more realistic)
marketing_spend.loc[marketing_spend["channel"] == "Digital", "spend_amount"] *= 1.4
marketing_spend.loc[marketing_spend["channel"] == "Organic", "spend_amount"] *= 0.4

marketing_spend.head()


,channel,spend_date,spend_amount
0,Digital,2024-01-01,49798.952
1,Digital,2024-01-02,50850.730
2,Digital,2024-01-03,36353.898
3,Digital,2024-01-04,29757.490
4,Digital,2024-01-05,20387.724


# STEP 4: Data Validation & Quality Checks

---

## 1. Data Quality Philosophy

The purpose of this step is to validate the integrity, consistency, and usability of the dataset before performing analytical calculations.  
Rather than assuming clean data, explicit checks are performed to identify missing values, inconsistencies, and anomalies that could impact business insights.

---

## 2. Key Validation Checks

The following quality checks are applied:

- Record count consistency across tables
- Missing and null value analysis
- Date consistency and logical ordering
- Duplicate and orphan record checks
- Outlier detection in monetary fields

These checks ensure analytical outputs are reliable and defensible.

---


## DATA QUALITY CHECKS

In [ ]:
tables = {
    "customers": customers,
    "applications": applications,
    "activations": activations,
    "transactions": transactions,
    "revenue": revenue,
    "marketing_spend": marketing_spend
}

for name, df in tables.items():
    print(f"{name}: {df.shape}")


customers: (150000, 5)
applications: (150000, 5)
activations: (105056, 3)
transactions: (600000, 5)
revenue: (300000, 3)
marketing_spend: (720, 3)


In [ ]:
# Missing Value Analysis

for name, df in tables.items():
    print(f"\n{name.upper()} - Missing Values")
    display(df.isna().mean().sort_values(ascending=False))



CUSTOMERS - Missing Values


,0
segment,0.030547
customer_id,0.000000
acquisition_channel,0.000000
state,0.000000
signup_date,0.000000



APPLICATIONS - Missing Values


,0
approval_date,0.299627
customer_id,0.000000
application_date,0.000000
application_id,0.000000
application_status,0.000000



ACTIVATIONS - Missing Values


,0
activation_date,0.24861
customer_id,0.00000
activation_status,0.00000



TRANSACTIONS - Missing Values


,0
transaction_id,0.0
customer_id,0.0
transaction_date,0.0
transaction_amount,0.0
merchant_category,0.0



REVENUE - Missing Values


,0
revenue_amount,0.01971
customer_id,0.00000
revenue_date,0.00000



MARKETING_SPEND - Missing Values


,0
channel,0.0
spend_date,0.0
spend_amount,0.0


In [ ]:
# Referential Integrity Checks

# Customers without applications
missing_applications = set(customers["customer_id"]) - set(applications["customer_id"])
print("Customers without applications:", len(missing_applications))

# Approved customers without activation records
approved_customers = set(applications.loc[
    applications["application_status"] == "Approved", "customer_id"
])

activation_customers = set(activations["customer_id"])

print("Approved customers without activation records:",
      len(approved_customers - activation_customers))


Customers without applications: 0
Approved customers without activation records: 0


In [ ]:
# Date Consistency Checks

# Approval before application (should be zero)
invalid_dates = applications[
    applications["approval_date"] < applications["application_date"]
]

print("Invalid approval dates:", len(invalid_dates))


Invalid approval dates: 0


In [ ]:
# Outlier Detection (Transactions & Revenue)

transactions.describe(percentiles=[0.95, 0.99])


,transaction_id,customer_id,transaction_date,transaction_amount
count,600000.000000,600000.000000,600000,600000.000000
mean,300000.500000,75056.006088,2024-03-30 09:55:36.479999744,55.338750
min,1.000000,1.000000,2024-01-01 00:00:00,0.000000
50%,300000.500000,75037.000000,2024-03-30 00:00:00,33.100000
95%,570000.050000,142535.000000,2024-06-19 00:00:00,174.340000
99%,594000.010000,148507.000000,2024-06-27 00:00:00,353.680800
max,600000.000000,149999.000000,2024-06-28 00:00:00,5807.250000
std,173205.225094,43317.334005,NaN,76.339194


In [ ]:
revenue.describe(percentiles=[0.95, 0.99])

,customer_id,revenue_date,revenue_amount
count,300000.000000,300000,294087.000000
mean,75070.979443,2024-03-30 06:50:29.759999744,1.103896
min,1.000000,2024-01-01 00:00:00,0.000000
50%,75067.000000,2024-03-30 00:00:00,0.630000
95%,142541.000000,2024-06-19 00:00:00,3.580000
99%,148519.020000,2024-06-27 00:00:00,7.430000
max,149999.000000,2024-06-28 00:00:00,68.850000
std,43358.729012,NaN,1.590997


## Outlier Analysis & Interpretation

### Transaction Amounts

Transaction value distributions show a strong right-skew, which is typical for credit card spending behavior.  
While the median transaction amount is relatively low, a small percentage of transactions contribute disproportionately to total volume.

Extreme transaction values are retained in the dataset, as they represent genuine high-spend behavior rather than data errors.  
For analytical stability, transaction values are capped at the 99th percentile during aggregation, while raw values are preserved for customer-level analysis.

---

### Revenue Amounts

Revenue per transaction follows a similar heavy-tailed distribution.  
Most customers generate modest revenue, while a small subset contributes significantly higher revenue, reinforcing the importance of customer quality in acquisition strategy.

Missing revenue values are retained in the raw dataset and excluded only when calculating revenue-specific KPIs.

---

### Business Implication

The observed distributions highlight that optimizing acquisition strategy should focus on attracting high-quality, high-retention customers rather than maximizing acquisition volume alone.


## Cleaning Decisions & Rationale

Based on validation results, the following decisions are applied:

- Records with missing activation dates are retained to preserve funnel drop-offs.
- Missing customer segment values are categorized as "Unknown" to avoid data loss.
- Transaction amounts equal to zero are retained to reflect failed or reversed transactions.
- Extreme outliers are capped only at the 99th percentile during analysis, not removed.
- Revenue records with missing values are excluded only for revenue-specific KPIs.

These decisions ensure analytical integrity while preserving real-world behavior.

---


In [ ]:
# Handle Missing Segments

customers["segment"] = customers["segment"].fillna("Unknown")


In [ ]:
#Create Analysis-Ready Views

transactions_clean = transactions.copy()
revenue_clean = revenue.dropna(subset=["revenue_amount"]).copy()


In [ ]:
# Cap Extreme Values (for analysis only)

cap_value = transactions_clean["transaction_amount"].quantile(0.99)
transactions_clean["transaction_amount_capped"] = transactions_clean["transaction_amount"].clip(upper=cap_value)


# STEP 5: Acquisition Funnel Analytics

---

## 1. Funnel Definition

The acquisition funnel represents the customer journey from initial application through approval and eventual card activation.

The funnel stages analyzed are:
1. Applications submitted
2. Applications approved
3. Cards activated

Analyzing conversion rates at each stage helps identify operational bottlenecks and optimization opportunities across acquisition channels.

---


In [ ]:
# Merge core funnel tables
funnel_df = (
    customers[["customer_id", "acquisition_channel"]]
    .merge(applications[["customer_id", "application_status"]], on="customer_id", how="left")
    .merge(activations[["customer_id", "activation_status"]], on="customer_id", how="left")
)

funnel_df.head()


,customer_id,acquisition_channel,application_status,activation_status
0,1,Digital,Approved,Activated
1,2,Organic,Approved,Activated
2,3,Partner,Approved,Activated
3,4,Referral,Approved,Not Activated
4,5,Digital,Approved,Activated


In [ ]:
funnel_df["applied"] = 1

funnel_df["approved"] = (funnel_df["application_status"] == "Approved").astype(int)

funnel_df["activated"] = (funnel_df["activation_status"] == "Activated").astype(int)

funnel_df.head()


,customer_id,acquisition_channel,application_status,activation_status,applied,approved,activated
0,1,Digital,Approved,Activated,1,1,1
1,2,Organic,Approved,Activated,1,1,1
2,3,Partner,Approved,Activated,1,1,1
3,4,Referral,Approved,Not Activated,1,1,0
4,5,Digital,Approved,Activated,1,1,1


In [ ]:
# OVERALL FUNNEL METRICS
funnel_summary = funnel_df.agg({
    "applied": "sum",
    "approved": "sum",
    "activated": "sum"
}).to_frame(name="count")

funnel_summary["conversion_rate"] = funnel_summary["count"] / funnel_summary.loc["applied", "count"]

funnel_summary


,count,conversion_rate
applied,150000,1.000000
approved,105056,0.700373
activated,78938,0.526253


In [ ]:
# FUNNEL BY ACQUISITION CHANNEL
channel_funnel = (
    funnel_df
    .groupby("acquisition_channel")
    .agg(
        applications=("applied", "sum"),
        approvals=("approved", "sum"),
        activations=("activated", "sum")
    )
)

channel_funnel["approval_rate"] = channel_funnel["approvals"] / channel_funnel["applications"]
channel_funnel["activation_rate"] = channel_funnel["activations"] / channel_funnel["approvals"]

channel_funnel = channel_funnel.sort_values("activation_rate", ascending=False)

channel_funnel


,applications,approvals,activations,approval_rate,activation_rate
acquisition_channel,,,,,
Partner,30193,21232,16000,0.703209,0.753580
Digital,67301,47024,35342,0.698712,0.751574
Referral,37664,26429,19828,0.701705,0.750236
Organic,14842,10371,7768,0.698760,0.749012


In [ ]:
# FUNNEL DROP-OFF ANALYSIS

channel_funnel["approval_dropoff"] = 1 - channel_funnel["approval_rate"]
channel_funnel["activation_dropoff"] = 1 - channel_funnel["activation_rate"]

channel_funnel[[
    "approval_rate",
    "activation_rate",
    "approval_dropoff",
    "activation_dropoff"
]]


,approval_rate,activation_rate,approval_dropoff,activation_dropoff
acquisition_channel,,,,
Partner,0.703209,0.753580,0.296791,0.246420
Digital,0.698712,0.751574,0.301288,0.248426
Referral,0.701705,0.750236,0.298295,0.249764
Organic,0.698760,0.749012,0.301240,0.250988


## Acquisition Funnel Performance — Business Interpretation

### Overall Funnel Health

The end-to-end acquisition funnel shows that approximately **70% of applications are approved**, while only **~53% of applicants ultimately activate** their cards.  
This indicates that while credit approval performance is relatively strong, **activation represents the largest point of customer drop-off** in the acquisition lifecycle.

The gap between approval and activation suggests onboarding friction or reduced customer intent post-approval, making activation optimization a high-impact opportunity for improving acquisition efficiency.

---

### Channel-Level Funnel Performance

Across acquisition channels, approval rates remain broadly consistent, ranging between **~69% and ~70%**, indicating that credit risk and underwriting criteria are applied uniformly across channels.

However, meaningful differences emerge when analyzing activation performance:

- **Partner and Digital channels** demonstrate slightly stronger activation rates (~75%), indicating higher customer intent or better onboarding effectiveness.
- **Referral and Organic channels**, while delivering steady application volumes, exhibit marginally lower activation rates, contributing to higher post-approval drop-offs.

Although the absolute differences appear modest, even small improvements in activation rates can translate into a significant increase in activated customers at scale.

---

### Funnel Drop-Off Insights

Activation drop-off rates (~24–25%) are consistently higher than approval drop-offs across all channels.  
This highlights that **customer engagement after approval is a more critical bottleneck than application screening**.

Potential drivers may include:
- Delays in card delivery or activation communication
- Suboptimal onboarding experiences
- Lower immediate perceived value post-approval

---

### Strategic Implications

- **Prioritize activation optimization** as the primary lever for improving acquisition ROI, rather than increasing top-of-funnel volume.
- **Sustain investment in Partner and Digital channels**, which demonstrate relatively stronger end-to-end funnel efficiency.
- **Target onboarding improvements** for Referral and Organic channels to reduce post-approval attrition.
- Introduce channel-specific activation interventions such as guided activation journeys, incentives, or personalized communication.

---

### Executive Takeaway

Improving activation rates by even **2–3 percentage points** would materially increase the number of revenue-generating customers without additional acquisition spend.  
As a result, activation-focused initiatives represent one of the highest ROI opportunities within the current acquisition strategy.


# STEP 6: Marketing Spend, CAC & Channel ROI Analysis

---

## 1. Objective

This analysis evaluates the efficiency of marketing spend across acquisition channels by linking channel-level investment to customer acquisition outcomes.

The goal is to understand:
- How much it costs to acquire an activated customer
- Which channels deliver the best balance of volume, quality, and cost efficiency
- Where marketing investments can be optimized to improve ROI

---


In [ ]:
# Aggregate Marketing Spend by Channel

channel_spend = (
    marketing_spend
    .groupby("channel", as_index=False)
    .agg(total_spend=("spend_amount", "sum"))
)

channel_spend


,channel,total_spend
0,Digital,7000736.512
1,Organic,1918797.296
2,Partner,5257413.050
3,Referral,4807065.050


In [ ]:
channel_metrics = (
    channel_funnel
    .reset_index()
    .merge(channel_spend, left_on="acquisition_channel", right_on="channel", how="left")
)

channel_metrics


,acquisition_channel,applications,approvals,activations,approval_rate,activation_rate,approval_dropoff,activation_dropoff,channel,total_spend
0,Partner,30193,21232,16000,0.703209,0.753580,0.296791,0.246420,Partner,5257413.050
1,Digital,67301,47024,35342,0.698712,0.751574,0.301288,0.248426,Digital,7000736.512
2,Referral,37664,26429,19828,0.701705,0.750236,0.298295,0.249764,Referral,4807065.050
3,Organic,14842,10371,7768,0.698760,0.749012,0.301240,0.250988,Organic,1918797.296


In [ ]:
# CAC Calculations

channel_metrics["CAC"] = (
    channel_metrics["total_spend"] / channel_metrics["activations"]
)

channel_metrics = channel_metrics.sort_values("CAC")

channel_metrics[[
    "acquisition_channel",
    "applications",
    "approvals",
    "activations",
    "approval_rate",
    "activation_rate",
    "total_spend",
    "CAC"
]]


,acquisition_channel,applications,approvals,activations,approval_rate,activation_rate,total_spend,CAC
1,Digital,67301,47024,35342,0.698712,0.751574,7000736.512,198.085465
2,Referral,37664,26429,19828,0.701705,0.750236,4807065.050,242.438221
3,Organic,14842,10371,7768,0.698760,0.749012,1918797.296,247.013040
0,Partner,30193,21232,16000,0.703209,0.753580,5257413.050,328.588316


In [ ]:
# CAC VS FUNNEL QUALITY

channel_metrics["cost_per_application"] = (
    channel_metrics["total_spend"] / channel_metrics["applications"]
)

channel_metrics["cost_per_approval"] = (
    channel_metrics["total_spend"] / channel_metrics["approvals"]
)

channel_metrics[[
    "acquisition_channel",
    "cost_per_application",
    "cost_per_approval",
    "CAC"
]]


,acquisition_channel,cost_per_application,cost_per_approval,CAC
1,Digital,104.021285,148.875819,198.085465
2,Referral,127.630232,181.885998,242.438221
3,Organic,129.281586,185.015649,247.013040
0,Partner,174.126885,247.617419,328.588316


## Marketing Spend & CAC — Business Interpretation

### Overall Spend Allocation

Marketing spend is unevenly distributed across acquisition channels, with **Digital** receiving the highest investment, followed by **Partner** and **Referral** channels.  
Despite lower total spend, **Organic** continues to contribute a meaningful volume of activated customers, indicating strong inherent intent but limited scalability.

---

## Customer Acquisition Cost (CAC) Comparison

Customer Acquisition Cost (CAC) varies significantly across acquisition channels:

- **Digital** delivers the lowest CAC (~198), driven by strong activation performance at scale.
- **Referral** and **Organic** channels exhibit moderate CAC (~242–247), reflecting stable funnel performance but lower cost efficiency.
- **Partner** channel shows the highest CAC (~329), despite comparable approval and activation rates.

This analysis highlights that cost efficiency is not solely driven by funnel conversion rates, but also by the level of marketing investment required to acquire customers.


---

### Cost vs Funnel Efficiency Trade-Off

While Partner and Digital channels demonstrate similar activation rates (~75%), their cost structures differ materially.  
The Partner channel requires significantly higher spend per application and per approval, reducing overall acquisition efficiency.

Digital, by contrast, achieves a more favorable balance between **scale, activation efficiency, and cost**, making it the most efficient growth driver in the current mix.

---

### Strategic Implications

- **Prioritize incremental investment in Digital acquisition**, where lower CAC and strong activation rates maximize return on spend.
- **Maintain Referral and Organic channels** as complementary acquisition sources, while exploring cost optimization opportunities.
- **Reassess Partner channel strategy**, focusing on improving targeting efficiency, renegotiating partnership costs, or refining partner selection criteria.

---

### Executive Recommendation

Reallocating a portion of Partner channel spend toward lower-CAC channels could increase the number of activated customers without increasing overall marketing budget.  
A CAC-driven acquisition strategy enables sustainable growth by balancing volume, customer quality, and cost efficiency.


# STEP 7: Revenue, Customer Quality & Early LTV Analysis

---

## 1. Objective

This analysis evaluates the quality of acquired customers by linking acquisition channels and customer segments to early-stage revenue generation.

Rather than focusing solely on acquisition cost, the objective is to understand which channels and segments deliver customers with stronger revenue potential and long-term value signals.

---


In [ ]:
# Aggregate Revenue at Customer Level

customer_revenue = (
    revenue_clean
    .groupby("customer_id", as_index=False)
    .agg(
        total_revenue=("revenue_amount", "sum"),
        revenue_events=("revenue_amount", "count")
    )
)

customer_revenue.head()


,customer_id,total_revenue,revenue_events
0,1,2.06,3
1,2,3.25,2
2,3,2.30,3
3,5,2.50,3
4,7,6.06,3


In [ ]:
customer_value = (
    customers[["customer_id", "acquisition_channel", "segment"]]
    .merge(customer_revenue, on="customer_id", how="left")
)

customer_value["total_revenue"] = customer_value["total_revenue"].fillna(0)
customer_value["revenue_events"] = customer_value["revenue_events"].fillna(0)

customer_value.head()


,customer_id,acquisition_channel,segment,total_revenue,revenue_events
0,1,Digital,Mass,2.06,3.0
1,2,Organic,Mass,3.25,2.0
2,3,Partner,Affluent,2.30,3.0
3,4,Referral,Affluent,0.00,0.0
4,5,Digital,Affluent,2.50,3.0


In [ ]:
# REVENUE QUALITY BY CHANNEL
channel_revenue_quality = (
    customer_value
    .groupby("acquisition_channel")
    .agg(
        customers=("customer_id", "nunique"),
        avg_revenue_per_customer=("total_revenue", "mean"),
        median_revenue=("total_revenue", "median"),
        pct_revenue_generating=("total_revenue", lambda x: (x > 0).mean())
    )
    .sort_values("avg_revenue_per_customer", ascending=False)
)

channel_revenue_quality


,customers,avg_revenue_per_customer,median_revenue,pct_revenue_generating
acquisition_channel,,,,
Partner,30193,2.179282,0.37,0.517570
Organic,14842,2.172557,0.28,0.511858
Referral,37664,2.172007,0.33,0.514284
Digital,67301,2.151391,0.31,0.512756


In [ ]:
# EARLY LTV PROXY (PRACTICAL & REALISTIC)

# Early LTV proxy = avg revenue per customer * activation rate
early_ltv = (
    channel_revenue_quality
    .merge(
        channel_funnel[["activation_rate"]],
        left_index=True,
        right_index=True
    )
)

early_ltv["early_LTV_proxy"] = (
    early_ltv["avg_revenue_per_customer"] * early_ltv["activation_rate"]
)

early_ltv.sort_values("early_LTV_proxy", ascending=False)


,customers,avg_revenue_per_customer,median_revenue,pct_revenue_generating,activation_rate,early_LTV_proxy
acquisition_channel,,,,,,
Partner,30193,2.179282,0.37,0.517570,0.753580,1.642262
Referral,37664,2.172007,0.33,0.514284,0.750236,1.629519
Organic,14842,2.172557,0.28,0.511858,0.749012,1.627270
Digital,67301,2.151391,0.31,0.512756,0.751574,1.616929


In [ ]:
# SEGMENT-LEVEL QUALITY ANALYSIS

segment_quality = (
    customer_value
    .groupby("segment")
    .agg(
        customers=("customer_id", "nunique"),
        avg_revenue=("total_revenue", "mean"),
        median_revenue=("total_revenue", "median")
    )
    .sort_values("avg_revenue", ascending=False)
)

segment_quality


,customers,avg_revenue,median_revenue
segment,,,
Mass,94450,2.167068,0.33
SME,14598,2.164777,0.35
Affluent,36370,2.163955,0.29
Unknown,4582,2.107671,0.34


## Revenue, Customer Quality & Early LTV — Business Interpretation

### Customer-Level Revenue Overview

Customer-level revenue analysis indicates that revenue generation is unevenly distributed, with a subset of customers contributing disproportionately to total revenue.  
While many customers generate modest early-stage revenue, a meaningful portion show repeated revenue events, signaling engagement and long-term value potential.

This reinforces the importance of evaluating customer quality beyond simple activation metrics.

---

### Revenue Quality by Acquisition Channel

Average revenue per customer remains relatively consistent across acquisition channels, ranging from approximately **$2.15 to $2.18**.  
However, differences emerge in **median revenue** and the **percentage of revenue-generating customers**, indicating variation in customer engagement depth.

- **Partner channel** exhibits the highest average and median revenue, suggesting stronger early customer value.
- **Referral and Organic channels** demonstrate stable revenue generation but lower median values, indicating a broader mix of low- and mid-value customers.
- **Digital channel**, while highly cost-efficient and scalable, shows slightly lower early revenue per customer compared to other channels.

These patterns suggest that volume-efficient channels do not always deliver the highest-value customers.

---

### Early LTV Proxy Insights

By combining average revenue per customer with activation rates, an early LTV proxy highlights meaningful differences in value contribution across channels:

- **Partner channel** ranks highest on early LTV, driven by strong activation performance and higher revenue per customer.
- **Referral and Organic channels** demonstrate comparable early LTV, reflecting balanced funnel efficiency and customer quality.
- **Digital channel**, despite delivering the lowest CAC, produces slightly lower early LTV, indicating a trade-off between cost efficiency and customer value.

This confirms that **low CAC alone is insufficient for optimizing long-term growth**.

---

### Strategic Implications

- Channels such as **Partner** deliver higher-value customers but at higher acquisition cost, requiring careful ROI evaluation.
- **Digital acquisition** remains critical for scale but should be complemented with value-based optimization strategies.
- **Referral and Organic channels** offer balanced performance and may benefit from targeted activation and engagement enhancements.

---

### Executive Recommendation

Acquisition strategy should shift from a cost-only optimization framework toward a **value-adjusted approach**, balancing CAC with early customer value.

Prioritizing channels and segments that maximize early LTV relative to acquisition cost will enable more sustainable, profitable growth while maintaining acquisition scale.
